cli/src/main/java/com/devonfw/tools/ide/repo/AbstractToolRepository.java

```diff
@@ -49,6 +50,9 @@ public Path download(String tool, String edition, VersionIdentifier version) {
     UrlDownloadFileMetadata metadata = getMetadata(tool, edition, version);
     VersionIdentifier resolvedVersion = metadata.getVersion();
     Set<String> urlCollection = metadata.getUrls();
+    if (context.isOffline()) {
+      throw new CliOfflineException("Not able to download " + tool + " because we are offline");
+    }
```

Comment:
This download method delegates to `FileAccessImpl.download` later (see below).
Why do we need to have this here?
Because we want to abort sooner and avoid waste (creating temporary path)? Or because we want to provide additional information?

Reply:
We can download Tools via the implementation in AbstractToolRepository.java or we can download plugins via the implementation in FileAccessImpl.java in both cases we need to cancel since we are offline. Additionaly the download method of FileAccessImpl is called from other methods directly, so we need to have the check in both methods.